src/shared/telemetry/telemetryPublisher.ts

```diff
@@ -0,0 +1,15 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { TelemetryEvent } from './telemetryEvent'
+
+export interface TelemetryPublisher {
+    init(): Promise<void>
+
+    enqueue(events: TelemetryEvent[]): any
```

Comment:
Nit: For convenience, consider using: `enqueue(...events: TelemetryEvent[]): any`.

Reply:
:+1: